app/presenters/dashboard.rb

```diff
@@ -14,6 +14,10 @@ def unsubmitted_exercises
         user.exercises.unsubmitted
       end
 
+      def unsubmitted_grouped_exercises
+        user.exercises.unsubmitted.group_by { |ex| ex.language }
```

Comment:
I literally just did some memoization refactors at work last week and am totally onboard with doing that here. I'll work on that (and figuring out the javascript stuff) this week. Thanks for the nitpick!